From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Date: Thu, 12 May 2016 22:25:32 +0800
Subject: Provide access to some of NM's interfaces to whoopsie.

Whoopsie is the crash database reporting daemon. It needs access to some of
the information NM keeps about devices to avoid sending data over the network
when connected to 3G or other systems that are potentially billable.
---
 src/core/org.freedesktop.NetworkManager.conf | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/core/org.freedesktop.NetworkManager.conf b/src/core/org.freedesktop.NetworkManager.conf
index 5c2af2e..f56c7fb 100644
--- a/src/core/org.freedesktop.NetworkManager.conf
+++ b/src/core/org.freedesktop.NetworkManager.conf
@@ -37,6 +37,19 @@
         <allow own="org.freedesktop.NetworkManager.dnsmasq"/>
         <allow send_destination="org.freedesktop.NetworkManager.dnsmasq"/>
     </policy>
+    <policy user="whoopsie">
+            <allow send_destination="org.freedesktop.NetworkManager"/>
+            <allow send_destination="org.freedesktop.NetworkManager"
+                   send_interface="org.freedesktop.DBus.Introspectable"/>
+            <allow send_destination="org.freedesktop.NetworkManager"
+                   send_interface="org.freedesktop.DBus.Properties"/>
+            <allow send_destination="org.freedesktop.NetworkManager"
+                   send_interface="org.freedesktop.NetworkManager"/>
+            <allow send_destination="org.freedesktop.NetworkManager"
+                   send_interface="org.freedesktop.NetworkManager.Connection.Active"/>
+            <allow send_destination="org.freedesktop.NetworkManager"
+                   send_interface="org.freedesktop.NetworkManager.Device"/>
+    </policy>
     <policy context="default">
         <deny own="org.freedesktop.NetworkManager"/>
 
